cmd/harmony/flags.go

```diff
@@ -244,6 +244,10 @@ func applyGeneralFlags(cmd *cobra.Command, config *harmonyConfig) {
 		config.General.NodeType = cli.GetStringFlagValue(cmd, legacyNodeTypeFlag)
 	}
 
+	if config.General.NodeType == nodeTypeExplorer {
+		config.General.IsArchival = true
```

Comment:
always override it?

Reply:
If flag is set by user, user's setting will override this option. Please check `cmd/harmony/flags.go:263`.

Also, a unit test is implemented to show the archival mode is overriden by user's option `cmd/harmony/flags_test.go:178`

Reply:
thanks for the explanation.